lib/asset/transferpart.js

```diff
@@ -30,9 +30,10 @@ class TransferPart {
      * @param {TransferAsset} transferAsset Transfer asset
      * @param {URL[]} targetUrls Target urls for this part
      * @param {DRange} contentRange Range for this part
-     * @param {Headers} [headers] Headers to send when storing content in the target urls
+     * @param {Headers} [sourceHeaders] Headers to send when interacting with the source
+     * @param {Headers} [targetHeaders] Headers to send when interacting with the target
      */
-    constructor(transferAsset, targetUrls, contentRange, headers) {
+    constructor(transferAsset, targetUrls, contentRange, sourceHeaders, targetHeaders) {
```

Comment:
That starts to be a lot of parameters... If you need more, consider using objects:
```
// httpSettings contains contentRange, sourceHeaders and targetHeaders, 
// possibly more properties if you would need more
constructor(transferAsset, targetUrls, httpSettings) { //... }
```